cpu/cc2538/radio/cc2538_rf_radio_ops.c

```diff
@@ -0,0 +1,471 @@
+#include <errno.h>
+#include <stdio.h>
+
+#include "net/gnrc.h"
+
+#include "cc2538_rf.h"
+#include "cc2538_rf_internal.h"
+
+#include "net/ieee802154/radio.h"
+
+#define CC2538_ACCEPT_FT_2_ACK     (1 << 5)
+
+#if IS_USED(MODULE_IEEE802154_RADIO_HAL)
+static const ieee802154_radio_ops_t cc2538_rf_ops;
+
+ieee802154_dev_t cc2538_rf_dev = {
+    .driver = &cc2538_rf_ops,
+};
+
+static uint8_t cc2538_min_be = 3;
+static uint8_t cc2538_max_be = 5;
+static int cc2538_retries = 4;
+
+static bool cc2538_cca_status;
+static bool cc2538_cca;
+
+static int _write(ieee802154_dev_t *dev, iolist_t *iolist)
+{
+    (void) dev;
+    int pkt_len = 0;
+    RFCORE_SFR_RFST = ISFLUSHTX;
+    rfcore_write_byte(0);
+
+    for (const iolist_t *iol = iolist; iol; iol = iol->iol_next) {
+        if (iol->iol_len) {
+            pkt_len += iol->iol_len;
+            rfcore_write_fifo(iol->iol_base, iol->iol_len);
+        }
+    }
+
+    /* Set first byte of TX FIFO to the packet length */
+    rfcore_poke_tx_fifo(0, pkt_len + CC2538_AUTOCRC_LEN);
+    return 0;
+}
+
+static int _confirm_transmit(ieee802154_dev_t *dev, ieee802154_tx_info_t *info)
+{
+    (void) dev;
+
+    if (RFCORE->XREG_FSMSTAT1bits.TX_ACTIVE != 0 || !(RFCORE_XREG_CSPCTRL & CC2538_CSP_MCU_CTRL_MASK)) {
+        return -EAGAIN;
+    }
+
+    if (info) {
+        if (cc2538_retries >= 0 && RFCORE_XREG_CSPZ == 0) {
+            info->status = TX_STATUS_MEDIUM_BUSY;
+        }
+        else {
+            info->status = TX_STATUS_SUCCESS;
+        }
+    }
+
+    return 0;
+}
+
+static int _request_transmit(ieee802154_dev_t *dev)
+{
+    (void) dev;
+
+    if (cc2538_retries < 0) {
+        RFCORE_SFR_RFST = ISTXON;
+    }
+    else {
+        cc2538_cca = false;
+
+        RFCORE_SFR_RFST = ISRXON;
+        /* Clear last program */
+        RFCORE_SFR_RFST = ISCLEAR;
+
+        /*  If the RSSI is not yet valid, busy loop */
+        RFCORE_SFR_RFST = SKIP_S_C |
+                          CC2538_CSP_SKIP_N_MASK |
+                          CC2538_CSP_SKIP_COND_RSSI;
+
+        RFCORE_SFR_RFST = LABEL;
+
+        RFCORE_SFR_RFST = RANDXY;
+        RFCORE_SFR_RFST = WAITX;
+
+        /* If CCA is valid, stop */
+        RFCORE_SFR_RFST = SKIP_S_C |
+                          (1 << CC2538_CSP_SKIP_INST_POS) |
+                          CC2538_CSP_SKIP_N_MASK |
+                          CC2538_CSP_SKIP_COND_CCA;
+
+        RFCORE_SFR_RFST = STOP;
+
+        RFCORE_SFR_RFST = DECZ;
+        RFCORE_SFR_RFST = INCMAXY | (cc2538_max_be & CC2538_CSP_INCMAXY_MAX_MASK);
+
+        RFCORE_SFR_RFST = RPT_C |
+                          CC2538_CSP_SKIP_N_MASK |
+                          CC2538_CSP_SKIP_COND_CSPZ;
```

Comment:
```suggestion
        RFCORE_SFR_RFST = RPT_C
                        | CC2538_CSP_SKIP_N_MASK
                        | CC2538_CSP_SKIP_COND_CSPZ;
```
looks a bit nicer IMHO, but's just my personal taste.

Reply:
done